docs/fundamentals/health-checks.md

```diff
@@ -110,6 +117,76 @@ builder.AddNpgsqlDbContext<MyDbContext>(
     static settings => settings.DisableHealthChecks  = true);
 ```
 
+## AppHost resource health checks
+
+AppHost resource health checks are different from the health check endpoints described earlier. These health checks are configured in the AppHost project and determine the readiness of resources from the orchestrator's perspective. They're particularly important for controlling when dependent resources start via the [`WaitFor`](orchestrate-resources.md#waiting-for-resources) functionality and are displayed in the Aspire dashboard.
+
+### Resource readiness with health checks
+
+When a resource has health checks configured, the AppHost uses them to determine if the resource is ready before starting dependent resources. If no health checks are registered for a resource, the AppHost waits for the resource to be in the <xref:Aspire.Hosting.ApplicationModel.KnownResourceStates.Running> state.
+
+### HTTP health checks for resources
+
+For resources that expose HTTP endpoints, you can add health checks that poll specific paths:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var catalogApi = builder.AddContainer("catalog-api", "catalog-api")
+                        .WithHttpEndpoint(targetPort: 8080)
+                        .WithHttpHealthCheck("/health");
+
+builder.AddProject<Projects.WebApp>("webapp")
+       .WithReference(catalogApi)
+       .WaitFor(catalogApi); // Waits for /health to return HTTP 200
+```
+
+The `WithHttpHealthCheck` method can also be applied to project resources:
+
+```csharp
+var backend = builder.AddProject<Projects.Backend>("backend")
+                     .WithHttpHealthCheck("/health");
+                     
+builder.AddProject<Projects.Frontend>("frontend")
+       .WithReference(backend)
+       .WaitFor(backend);
+```
+
```

Comment:
Does it make sense to display and communicate the expected shape of a health check endpoint? What does that look like, is it just an:

```http
GET /health HTTP/1.1
Host: localhost:5713

HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8

{
  "status": "Healthy"
}
```